pkg/operator/controller/extension/admission/admission.go

```diff
@@ -208,11 +212,19 @@ func (d *deployment) createOrUpdateAdmissionVirtualClusterResources(ctx context.
 		}
 	}
 
-	renderedChart, err := virtualClusterClientSet.ChartRenderer().RenderArchive(archive, extension.Name, v1beta1constants.GardenNamespace, utils.MergeMaps(helmValues, gardenerValues))
+	namespace := virtualNamespace(extension)
+	renderedChart, err := virtualClusterClientSet.ChartRenderer().RenderArchive(archive, extension.Name, namespace.Name, utils.MergeMaps(helmValues, gardenerValues))
 	if err != nil {
 		return fmt.Errorf("failed rendering Helm chart: %w", err)
 	}
 
+	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, virtualClusterClientSet.Client(), namespace, func() error {
```

Comment:
I would move the creation of the namespace before rendering the extension as setting the owner references depend on it existing.

Reply:
The owner reference is created not created by the helm chart but from the admission-controller of the extension itself. Thus, the order is not important for that matter.
I put the namespace creation after the chart rendering that we don't create an unused namespace in case the Helm rendering fails.

Reply:
The namespace is now part of the managed resource ([ref](https://github.com/gardener/gardener/pull/10585#discussion_r1789243334)).